cros_gralloc/cros_gralloc_driver.cc

```diff
@@ -248,15 +250,40 @@ cros_gralloc_driver::cros_gralloc_driver()
 					video_idx);
 		}
 
+		if ((virtio_node_idx != -1) && (virtio_node_idx != renderer_idx)) {
+			drv_kms_ = drv_create(node_fd[virtio_node_idx]);
+			if (!drv_kms_) {
+				drv_loge("Failed to create driver for the virtio-gpu device with card id %d\n",
+                                          virtio_node_idx);
+				close(node_fd[virtio_node_idx]);
+				drv_kms_ = drv_render_;
+                        }
+		} else {
+			drv_kms_ = drv_render_;
+		}
+
 		// Init drv
 		DRV_INIT(drv_render_, gpu_grp_type, renderer_idx)
 		if (video_idx != renderer_idx)
 			DRV_INIT(drv_video_, gpu_grp_type, video_idx)
+		if ((virtio_node_idx != -1) && (virtio_node_idx != renderer_idx))
+			DRV_INIT(drv_kms_, gpu_grp_type, virtio_node_idx)
+		if (drv_kms_ && (virtio_node_idx != renderer_idx)) {
+			bool virtiogpu_with_blob = drv_virtgpu_with_blob(drv_kms_);
```

Comment:
With this change we will allocate scan-out buffers on those virtio-GPU devices that support blob but doesn't support QUERY_DEV, is that what we want?

Looks like you want to distinguish normal virtio-GPU devices from ivshmem-backed devices, but I think `drv_virtgpu_with_blob` is not enough. If I understand correctly, we will be able to query the device name with a new ioctl if QUERY_DEV feature is present. So why don't we call this ioctl and determine whether it's ivishmem device according to its name?